examples/shell/shell_common/cmd_ping_responder.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "shell/streamer.h"
+#include <ChipShellCollection.h>
+
+#include <platform/CHIPDeviceLayer.h>
+#include <protocols/echo/Echo.h>
+#include <protocols/secure_channel/PASESession.h>
+#include <shell/shell_core.h>
+#include <support/ErrorStr.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/raw/TCP.h>
+#include <transport/raw/UDP.h>
+
+using chip::Shell::shell_command_t;
+using chip::Shell::streamer_get;
+using chip::Shell::streamer_printf;
+
+namespace {
+
+class EchoServerCmd
+{
+public:
+    EchoServerCmd() = default;
+
+    CHIP_ERROR Start(bool tcp)
+    {
+        if (mConnected)
+        {
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+        chip::Optional<chip::Transport::PeerAddress> peer(chip::Transport::Type::kUndefined);
+        mAdminInfo = mAdmins.AssignAdminId(kAdminId, chip::kTestDeviceNodeId);
+        if (mAdminInfo == NULL)
+        {
+            return CHIP_ERROR_NO_MEMORY;
+        }
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+        mTcp = tcp;
+        if (tcp)
+        {
+            ReturnErrorOnFailure(mTCPManager.Init(chip::Transport::TcpListenParameters(&chip::DeviceLayer::InetLayer)
+                                                      .SetAddressType(chip::Inet::kIPAddressType_IPv4)));
+            ReturnErrorOnFailure(
+                mSessionManager.Init(chip::kTestDeviceNodeId, &chip::DeviceLayer::SystemLayer, &mTCPManager, &mAdmins));
+        }
+#endif // INET_CONFIG_ENABLE_TCP_ENDPOINT
```

Comment:
if 'tcp' and TCP is not configured, should we return a invalid argument error instead of just using UDP?